core/src/main/scala/org/http4s/TransferCoding.scala

```diff
@@ -18,23 +18,52 @@
  */
 package org.http4s
 
-import org.http4s.syntax.string._
+import cats.{Order, Show}
+import cats.data.NonEmptyList
 import org.http4s.util._
+import org.http4s.parser.{Http4sParser, Rfc2616BasicRules}
 
-final case class TransferCoding private (coding: CaseInsensitiveString) extends Renderable {
+sealed abstract case class TransferCoding private (coding: String) extends Renderable {
   override def render(writer: Writer): writer.type = writer.append(coding.toString)
 }
 
-object TransferCoding extends Registry {
-  type Key = CaseInsensitiveString
-  type Value = TransferCoding
+object TransferCoding {
+  private class TransferCodingImpl(coding: String) extends TransferCoding(coding)
 
-  implicit def fromKey(k: CaseInsensitiveString): TransferCoding = new TransferCoding(k)
+  // https://www.iana.org/assignments/http-parameters/http-parameters.xml#transfer-coding
+  val chunked: TransferCoding = new TransferCodingImpl("chunked")
+  val compress: TransferCoding = new TransferCodingImpl("compress")
+  val deflate: TransferCoding = new TransferCodingImpl("deflate")
+  val gzip: TransferCoding = new TransferCodingImpl("gzip")
+  val identity: TransferCoding = new TransferCodingImpl("identity")
+
+  /**
+    * Parse a list of Transfer Coding entries
+    */
+  def parseList(s: String): ParseResult[NonEmptyList[TransferCoding]] =
+    new Http4sParser[NonEmptyList[TransferCoding]](s, "Invalid Transfer Coding")
+    with Rfc2616BasicRules {
+      private def codingRule = rule {
```

Comment:
We've lost the ability to create a custom `TransferCoding`.  Which probably rarely if ever happens in the wild, and isn't understood by any of our server or client implementations, so it's not a big deal.  But it's legal in the spec.

Reply:
Right, was that the reason to have it as a `Registry`? To support custom ones?